File: scripts/generate_icon.dart
import 'dart:io';
import 'dart:ui' as ui;
import 'package:flutter/material.dart';

void main() async {
  // Initialize Flutter bindings
  WidgetsFlutterBinding.ensureInitialized();
  
  print('🎨 Generating Diaspora Handbook icon...');
  
  // Icon size
  const size = 1024.0;
  
  // Create a picture recorder
  final recorder = ui.PictureRecorder();
  final canvas = Canvas(recorder);
  
  // Background gradient (matching app theme)
  final backgroundGradient = LinearGradient(
    begin: Alignment.topLeft,
    end: Alignment.bottomRight,
    colors: [
      Color(0xFF1a1a2e), // Dark blue
      Color(0xFF16213e), // Darker blue
      Color(0xFF0f3460), // Navy blue
    ],
  );
  
  final backgroundPaint = Paint()
    ..shader = backgroundGradient.createShader(
      Rect.fromLTWH(0, 0, size, size),
    );
  
  // Draw background with rounded corners
  final rrect = RRect.fromRectAndRadius(
    Rect.fromLTWH(0, 0, size, size),
    Radius.circular(size * 0.22), // iOS-style rounded corners
  );
  canvas.drawRRect(rrect, backgroundPaint);
  
  // Draw decorative sun rays in top left (subtle)
  final sunPaint = Paint()
    ..color = Color(0xFFFFD700).withOpacity(0.15)
    ..style = PaintingStyle.fill;
  
  for (int i = 0; i < 8; i++) {
    final angle = (i * 45.0) * (3.14159 / 180.0);
    final path = Path();
    path.moveTo(size * 0.15, size * 0.15);
    path.lineTo(
      size * 0.15 + 100 * (i % 2 == 0 ? 1.2 : 0.8) * (angle * 0.1).cos(),
      size * 0.15 + 100 * (i % 2 == 0 ? 1.2 : 0.8) * (angle * 0.1).sin(),
    );
    canvas.drawPath(path, sunPaint);
  }
  
  // Add a subtle border/glow effect
  final borderPaint = Paint()
    ..color = Color(0xFFFFD700).withOpacity(0.3)
    ..style = PaintingStyle.stroke
    ..strokeWidth = 4;
  canvas.drawRRect(rrect, borderPaint);
  
  // Draw "DH" text
  final textStyle = TextStyle(
    fontSize: size * 0.5, // Large text
    fontWeight: FontWeight.w900,
    letterSpacing: -20,
  );
  
  // Draw "D" in purple
  final dPainter = TextPainter(
    text: TextSpan(
      text: 'D',
      style: textStyle.copyWith(
        foreground: Paint()
          ..shader = LinearGradient(
            colors: [
              Color(0xFF9B59B6), // Purple
              Color(0xFF8E44AD), // Darker purple
            ],
          ).createShader(Rect.fromLTWH(0, 0, size * 0.3, size * 0.6)),
      ),
    ),
    textDirection: TextDirection.ltr,
  );
  dPainter.layout();
  
  // Draw "H" in gold/yellow
  final hPainter = TextPainter(
    text: TextSpan(
      text: 'H',
      style: textStyle.copyWith(
        foreground: Paint()
          ..shader = LinearGradient(
            colors: [
              Color(0xFFFFD700), // Gold
              Color(0xFFFFA500), // Orange-gold
            ],
          ).createShader(Rect.fromLTWH(0, 0, size * 0.3, size * 0.6)),
      ),
    ),
    textDirection: TextDirection.ltr,
  );
  hPainter.layout();
  
  // Calculate total width and center position
  final totalWidth = dPainter.width + hPainter.width - 20;
  final startX = (size - totalWidth) / 2;
  final startY = (size - dPainter.height) / 2;
  
  // Draw shadow for depth
  final shadowPaint = Paint()
    ..color = Colors.black.withOpacity(0.5)
    ..maskFilter = MaskFilter.blur(BlurStyle.normal, 20);
  
  canvas.drawRRect(
    RRect.fromRectAndRadius(
      Rect.fromLTWH(startX + 10, startY + 10, totalWidth, dPainter.height),
      Radius.circular(20),
    ),
    shadowPaint,
  );
  
  // Paint the letters
  dPainter.paint(canvas, Offset(startX, startY));
  hPainter.paint(canvas, Offset(startX + dPainter.width - 20, startY));
  
  // Add a subtle tagline at bottom
  final taglinePainter = TextPainter(
    text: TextSpan(
      text: 'HOMECOMING GUIDE',
      style: TextStyle(
        fontSize: size * 0.04,
        fontWeight: FontWeight.w600,
        color: Color(0xFFFFD700).withOpacity(0.8),
        letterSpacing: 8,
      ),
    ),
    textDirection: TextDirection.ltr,
  );
  taglinePainter.layout();
  taglinePainter.paint(
    canvas,
    Offset((size - taglinePainter.width) / 2, size * 0.85),
  );
  
  // Convert to image
  final picture = recorder.endRecording();
  final image = await picture.toImage(size.toInt(), size.toInt());
  final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
  
  if (byteData == null) {
    print('❌ Failed to generate image');
    exit(1);
  }
  
  // Save to file
  final file = File('assets/icon.png');
  await file.writeAsBytes(byteData.buffer.asUint8List());
  
  print('✅ Icon generated successfully: assets/icon.png');
  print('   Size: ${size.toInt()}x${size.toInt()} pixels');
  print('   Colors: Purple "D" + Gold "H"');
  print('');
  print('Next steps:');
  print('  1. Run: flutter pub run flutter_launcher_icons');
  print('  2. Run: flutter clean && flutter run');
}

extension on double {
  double cos() => this * 1.0; // Simplified for script
  double sin() => this * 1.0;
}

File: scripts/generate_icon.py
#!/usr/bin/env python3
"""
Diaspora Handbook Icon Generator
Creates a custom app icon with "DH" (D=purple, H=yellow/gold)
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon():
    print("🎨 Generating Diaspora Handbook icon...")
    
    # Icon size
    size = 1024
    
    # Create image with WHITE background
    img = Image.new('RGB', (size, size), color='#FFFFFF')
    draw = ImageDraw.Draw(img)
    
    # Draw subtle sun rays in corner (decorative) - light gold on white
    import math
    ray_img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    ray_draw = ImageDraw.Draw(ray_img)
    
    for i in range(8):
        angle = i * 45
        x1, y1 = size * 0.15, size * 0.15
        length = 150 if i % 2 == 0 else 100
        x2 = x1 + length * math.cos(math.radians(angle))
        y2 = y1 + length * math.sin(math.radians(angle))
        # Light gold rays on white background
        ray_draw.line([(x1, y1), (x2, y2)], fill=(255, 215, 0, 60), width=10)
    
    img = Image.alpha_composite(img.convert('RGBA'), ray_img).convert('RGB')
    draw = ImageDraw.Draw(img)
    
    # Add rounded corners (iOS style)
    mask = Image.new('L', (size, size), 0)
    mask_draw = ImageDraw.Draw(mask)
    mask_draw.rounded_rectangle([(0, 0), (size, size)], radius=int(size * 0.22), fill=255)
    
    # Apply mask
    output = Image.new('RGB', (size, size), (255, 255, 255))
    output.paste(img, (0, 0))
    img_with_mask = Image.new('RGBA', (size, size))
    img_with_mask.paste(output, (0, 0))
    img_with_mask.putalpha(mask)
    
    # Convert back to RGB for final output
    final_img = Image.new('RGB', (size, size), (255, 255, 255))
    final_img.paste(img_with_mask, (0, 0), img_with_mask)
    draw = ImageDraw.Draw(final_img)
    
    # Try to use a bold system font
    try:
        # Try different font options
        font_size = int(size * 0.55)
        font_paths = [
            '/System/Library/Fonts/Helvetica.ttc',
            '/System/Library/Fonts/Supplemental/Arial Bold.ttf',
            '/Library/Fonts/Arial Bold.ttf',
            'Arial',
        ]
        
        font = None
        for font_path in font_paths:
            try:
                if os.path.exists(font_path):
                    font = ImageFont.truetype(font_path, font_size)
                    break
            except:
                continue
        
        if not font:
            # Fallback to default
            font = ImageFont.load_default()
            font_size = 200  # Approximate size for default font
            
    except Exception as e:
        print(f"Warning: Could not load custom font: {e}")
        font = ImageFont.load_default()
        font_size = 200
    
    # Draw "D" in purple
    d_color = (155, 89, 182)  # Purple #9B59B6
    h_color = (255, 215, 0)   # Gold #FFD700
    
    # Calculate text positioning
    text = "DH"
    
    # Get text bounding box
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    # Center position
    x = (size - text_width) // 2
    y = (size - text_height) // 2 - 40  # Slightly above center
    
    # Draw shadow for depth (lighter shadow for white background)
    shadow_offset = 6
    draw.text((x + shadow_offset, y + shadow_offset), "D", font=font, fill=(200, 200, 200))
    draw.text((x + text_width//2 + shadow_offset, y + shadow_offset), "H", font=font, fill=(200, 200, 200))
    
    # Draw the letters
    draw.text((x, y), "D", font=font, fill=d_color)
    
    # Get width of "D" to position "H"
    d_bbox = draw.textbbox((x, y), "D", font=font)
    d_width = d_bbox[2] - d_bbox[0]
    
    draw.text((x + d_width - 30, y), "H", font=font, fill=h_color)  # Overlap slightly
    
    # Add tagline
    try:
        tagline_font_size = int(size * 0.035)
        tagline_font = ImageFont.truetype(font_paths[0], tagline_font_size) if os.path.exists(font_paths[0]) else font
    except:
        tagline_font = font
    
    tagline = "HOMECOMING GUIDE"
    tagline_bbox = draw.textbbox((0, 0), tagline, font=tagline_font)
    tagline_width = tagline_bbox[2] - tagline_bbox[0]
    tagline_x = (size - tagline_width) // 2
    tagline_y = int(size * 0.85)
    
    # Draw tagline in darker color for visibility on white
    draw.text((tagline_x, tagline_y), tagline, font=tagline_font, fill=(155, 89, 182))
    
    # Draw subtle border (lighter for white background)
    border_width = 3
    draw.rounded_rectangle(
        [(border_width, border_width), (size - border_width, size - border_width)],
        radius=int(size * 0.22),
        outline=(230, 230, 230),
        width=border_width
    )
    
    # Save the icon
    output_path = 'assets/icon.png'
    final_img.save(output_path, 'PNG', quality=100)
    
    print(f"✅ Icon generated successfully: {output_path}")
    print(f"   Size: {size}x{size} pixels")
    print(f"   Colors: Purple 'D' (#9B59B6) + Gold 'H' (#FFD700)")
    print("")
    print("Next steps:")
    print("  1. Run: flutter pub run flutter_launcher_icons")
    print("  2. Run: flutter clean && flutter run")

if __name__ == "__main__":
    create_icon()

